File: src/codes/Users_Manual/Tutorial/move_focus/hard.cs
~using AUTD3Sharp;
~using AUTD3Sharp.Utils;
~using AUTD3Sharp.Link;
~using AUTD3Sharp.Gain;
~using AUTD3Sharp.Modulation;
~using static AUTD3Sharp.Units;
~using var autd = Controller.Open([new AUTD3(pos: Point3.Origin, rot: Quaternion.Identity)], new Nop());
autd.Send(new Sine(
    freq: 150u * Hz,
    option: new SineOption()
));

var center = autd.Center() + new Vector3(0.0f, 0.0f, 150.0f);
autd.Send(new FociSTM(
    foci: [
        center + new Vector3(20.0f, 0.0f, 0.0f),
        center - new Vector3(20.0f, 0.0f, 0.0f),
    ],
    config: 0.5f * Hz
    // config: Duration.FromSecs(2)
    // config: new SamplingConfig(1.0f * Hz)
    // config: new SamplingConfig(Duration.FromSecs(1))
));

File: src/codes/Users_Manual/Tutorial/move_focus/hard.py
~import numpy as np
~from pyautd3 import AUTD3, Controller, Hz, Sine, SineOption, FociSTM
~from pyautd3.link.nop import Nop
~autd = Controller.open([AUTD3(pos=[0.0, 0.0, 0.0], rot=[1, 0, 0, 0])], Nop())
autd.send(
    Sine(
        freq=150 * Hz,
        option=SineOption(),
    )
)

center = autd.center() + np.array([0.0, 0.0, 150.0])
autd.send(
    FociSTM(
        foci=[
            center + np.array([20.0, 0.0, 0.0]),
            center - np.array([20.0, 0.0, 0.0]),
        ],
        config=0.5 * Hz,
        # config=Duration.from_secs(2),
        # config=SamplingConfig(1.0 * Hz),
        # config=SamplingConfig(Duration.from_secs(1)),
    )
)